### ISSUE TITLE
Inform of role conflict issues in aria-required-parent

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3495]
Axe-core correctly reports the following as an issue under `aria-required-parent`. The difficulty with it though, is that this looks like it should work. I think we need some custom message on this rule to communicate which attribute cause the conflict resolution, due to which the `role="none"` is ignored in some browsers:

```html
<ul role="menubar">
  <li role="none" tabindex="-1">
    <div role="menuitem">Hello world</div>
  </li>
</ul>
````

**Comment by WilcoFiers** [comment]
Here's another case that is sort of confusing for a similar reason:

```html
<ul role="menubar">
  <li>
    <div role="menuitem">Hello world</div>
  </li>
</ul>
```

This is flagged by the `listitem` rule as not having a `<ul>` or `<ol>` parent, which it clearly has. The message here should instead be that the parent has an incorrect role.
